src/istore_agg.c

```diff
@@ -3,18 +3,33 @@
-                                                                                             \
-    if (PG_ARGISNULL(1) && PG_ARGISNULL(0)) PG_RETURN_NULL();                                \
-                                                                                             \
-    state = PG_ARGISNULL(0) ? state_init(agg_context) : (ISAggState *) PG_GETARG_POINTER(0); \
-                                                                                             \
-    if (PG_ARGISNULL(1)) PG_RETURN_POINTER(state);                                           \
+#define BIG_ISTORE 1
+#define ISTORE 0
+
+#define RETURN_AGG_TRANSFN(_istore, _agg)                                                        \
+    do {                                                                                         \
+        MemoryContext  agg_context;                                                              \
+        ISAggState    *state;                                                                    \
+                                                                                                 \
+        if (!AggCheckCallContext(fcinfo, &agg_context))                                          \
+            elog(ERROR, "aggregate function called in non-aggregate context");                   \
+                                                                                                 \
+        if (PG_ARGISNULL(1) && PG_ARGISNULL(0)) PG_RETURN_NULL();                                \
+                                                                                                 \
+        state = PG_ARGISNULL(0) ? state_init(agg_context) : (ISAggState *) PG_GETARG_POINTER(0); \
+                                                                                                 \
+        if (PG_ARGISNULL(1)) PG_RETURN_POINTER(state);                                           \
+                                                                                                 \
+        switch (_istore)                                                                         \
+        {                                                                                        \
+            case BIG_ISTORE:                                                                     \
+                PG_RETURN_POINTER(bigistore_agg_internal(state, PG_GETARG_BIGIS(1), _agg));      \
+            case ISTORE:                                                                         \
+                PG_RETURN_POINTER(istore_agg_internal(state, PG_GETARG_IS(1), _agg));            \
+            default:                                                                             \
+                elog(ERROR, "unexpected istore type passed.");                                   \
+        }                                                                                        \
+    } while(0)
```

Comment:
i'm not 100% sure about this, but
1. the compiler should either error or not at all, since the case branches stay the same (syntactically speaking here) no matter what the input
2. the compiler probably doesn't because it realizes that the default branch is not taken
3. if there is no return statement in a function having a return value, the behavior is to return the value returned from the last function call. 

To catch this illegal assignment at compile time, one would have to enable it `-Werror=return-type` AFAICT.

The easiest way to go is to use a compile-time assertion like this:

```C
StaticAssertStmt(_istore == 0 || _istore == 1, "illegal istore type");
```